primo/data_parser/well_data.py

```diff
@@ -0,0 +1,1149 @@
+            )
+
+    def convert_data_to_binary(self, col_name: str):
+        """
+        Converts 1/True/"Yes"/"Y" -> 1 and 0/False/"No"/"N" -> 0.
+        The strings "Yes", "Y", "No", and "N" are case-insensitive i.e.,
+        "yes", "y", "no", "n" are also acceptable.
+
+        Parameters
+        ----------
+        col_name : str
+            Name of the column
+
+        Raises
+        ------
+        ValueError
+            If the column contains non-boolean-type data
+        """
+        # First convert the data to True or False
+        try:
+            self.data[col_name] = self.data[col_name].apply(Bool)
+        except ValueError as excp:
+            raise ValueError(
+                f"Column {col_name} is expected to contain boolean-type "
+                f"data. Received a non-boolean value for some/all rows."
+            ) from excp
+
+        # Now convert the data to binary
+        self.data[col_name] = self.data[col_name].apply(int)
+
+    def check_data_in_range(self, col_name: str, lb: float, ub: float):
+        """
+        Utility to check if all the values in a column are within
+        a valid interval.
+
+        Parameters
+        ----------
+        col_name : str
+            Name of the column
+        lb : float
+            Lower bound for the value
+        ub : float
+            Upper bound for the value
+
+        Raises
+        ------
+        ValueError
+            If one or more values in the column lie outside the valid range
+        """
+        # Check the validity of latitude and longitude
+        valid_data = self.data[col_name].apply(lambda val: lb <= val <= ub)
+        invalid_data = list(self.data[~valid_data].index)
+
+        if len(invalid_data) > 0:
+            msg = (
+                f"Values in column {col_name} are expected to be in the interval "
+                f"{[lb, ub]}. However, the value in one or more rows "
+                f"lies outside the interval."
+            )
+            raise_exception(msg, ValueError)
+
+    def is_data_numeric(self, col_name: str):
+        """
+        Checks if all the cells in a column are numeric.
+
+        Parameters
+        ----------
+        col_name : str
+            Name of the column
+        """
+        # There are two ways to do this:
+        # The current approach is a safe option since it also works for mixed
+        # data types, but it is a bit messy.
+        # Alternatively, one can use pd.api.types.is_numeric_dtype() method.
+        # This is quite compact, but this has risks. In the future, pandas will not
+        # allow modification of cells in a column to a different dtype. Then, it will
+        # be safe to use the pd.api.types.is_numeric_dtype() method.
+        flag = True
+        non_numeric_rows = []
+        for i in self:
+            # Record the row number if it is an empty cell
+            if pd.isnull(self.data.loc[i, col_name]):
```

Comment:
I think the logic here is sound, but perhaps a more succinct way to write it would be `if pd.isnull(self.data.loc[i, col_name]) or not pd.api.types.is_number(self.data.loc[i, col_name]):
    non_numeric_rows.append(i)`

Reply:
Made the change.